exoctk/phase_constraint_overlap/phase_constraint_overlap.py

```diff
@@ -100,14 +100,20 @@ def phase_overlap_constraint(target_name, period=None, t0=None, obs_duration=Non
             The maximum phase constraint. '''
 
     if obs_duration == None:
-        if period == None:
-            data = get_target_data(target_name)
-            
-            period = data['orbital_period']
-            transit_dur = data['transit_duration']
-            t0 = data['transit_time']
-
+        data = get_target_data(target_name)
+        transit_dur = data['transit_duration']
         obs_duration = calculate_obsDur(transit_dur)
+    
+    if period == None:
+        data = get_target_data(target_name)
+        period = data['orbital_period']
+    
+    if obs_duration == None and period == None:
```

Comment:
Is this combo needed? Won't it just repeat what is done above?

Reply:
Yeah, I want to rework this in the future for the other `if/else` blocks but I don't have time right now. I will remove the duplicate block for now.

Reply:
Since there are only 3 params needed to calculate the minimum and maximum phase, I went ahead and took care of the cases where the `period` or if the `observation_duration` is none. It will reflect in my next push